src/Compilers/Lua/Portable/SymbolDisplay/ObjectDisplay.cs

```diff
@@ -282,6 +282,19 @@ public static string FormatLiteral(long value, ObjectDisplayOptions options, Cul
             }
         }
 
+        public static string FormatLiteral(ulong value, ObjectDisplayOptions options, CultureInfo? cultureInfo = null)
+        {
+            if (options.IncludesOption(ObjectDisplayOptions.UseHexadecimalNumbers))
+            {
+                return value.ToString("D", GetFormatCulture(cultureInfo));
+            }
+            else
+            {
+                return value.ToString("X", GetFormatCulture(cultureInfo));
+            }
+        }
```

Comment:
This should add the `ULL` suffix to the number otherwise the wrong literals will be getting generated.